src/rollback.rs

```diff
@@ -45,3 +52,37 @@ impl<'w, 's, 'a> AddRollbackCommandExtension for EntityCommands<'w, 's, 'a> {
         self
     }
 }
+
+/// A [`Resource`] which provides methods for stable ordering of [`Rollback`] flags.
+#[derive(Resource, Default)]
+pub struct RollbackOrdered {
+    order: HashMap<Rollback, usize>,
+    sorted: Vec<Rollback>,
+}
+
+impl RollbackOrdered {
+    /// Register a new [`Rollback`] for explicit ordering.
+    fn push(&mut self, rollback: Rollback) -> &mut Self {
+        self.sorted.push(rollback);
+        self.sorted.sort_unstable();
```

Comment:
Items are never removed from these collections, which means as the game progresses and more and more entities have been added and despawned, spawning new entities will not only leak memory, but become more expensive as the cost of the sort grows.

Reply:
Good point. I've replaced the `sort_unstable` with a back-first bubble sort. Since `Rollback` entities should be created in order anyway (the first `Rollback` will be for the first entity, etc.), this should entirely avoid sorting in the vast majority of cases, and only take `n` swaps in the worst case.

As for garbage collecting dead `Rollback`'s out of this collection, I do agree that it would be a good idea, but I'm not sure what the best solution here is. I think it would be to compare the the `Rollback`'s contained in snapshots against `RollbackOrdered` and remove any that are no longer referenced, but I think that process would be fairly expensive to run every frame.

Since a pair of `Rollback` values (one in `sorted` and another in `order`) and an index as a `usize` take less than 24 bytes, it would take over 40,000 rollback entities before this object would hit 1MB of memory consumption, so the rate of lost memory would be incredibly slow.